amplify/backend/api/colonycdapp/schema.graphql

```diff
@@ -501,4 +513,99 @@ type ColonyAction @model {
   toDomain: Domain @hasOne(fields: ["toDomainId"])
   fundamentalChainId: Int
   newColonyVersion: Int
+  roles: ColonyActionRoles
+
+  # Required since some actions might have multiple event entries in the list
+  # which need to be iterated over and exposed individually
+  #
+  # This should be a list/array, but this is yet another thing Amplify is bad at
+  # The official response from them is to use DynamoDB directly:
+  # https://stackoverflow.com/questions/50992106/append-item-to-list-using-aws-appsync-to-dynamodb
+  # Which is just freaking great!
+  #
+  # As such, I've decided to simplify this, an just pass a stringified JSON around
+  # since this is for display purpouses only.
+  #
+  # If anyone has a better idea on how to handle this, I'll gladly change it!
+  #
+  # When / if you decide to use it, just be mindful of what format other people
+  # are using, and follow suit.
+  #
+  # Roles Types:
+  # [{ id: String, type: String, role: Number, setTo: boolean }]
```

Comment:
Circling back to it, I realised I actually didn't look into individual events as an actions concept at all. Should we come up with a solution that caters for all actions?
Otherwise maybe this should be renamed to be more roles-specific?

Reply:
Yes we should, I'm not particularly happy with my implementation here, I just did it since I wanted something fast to get it done already :)

Reply:
That's why I asked @willm30 about his implementation on Motions, as to try to align either this to his, or come up with a middle ground.